File: FrontEnd/src/app/data/position/position.model.ts
import { CategoryPosition } from '../categoryPosition/category-position.model';
import { Company } from '../company/company.model';
import { Language } from '../language/language.model';
import { Level } from '../level/level.model';
import { Recruiter } from '../recruiter/recruiter.model';
import { Requirements } from '../requirements/requirements.model';

export class Position {
	positionId?: string;
	positionName?: string;
	description?: string;
	imageURL?: string;
	minSalary?: number;
	maxSalary?: number;
	maxHiringQty?: number;
	startDate?: Date;
	endDate?: Date;
	company?: Company;
	language?: Language;
	recruiterId?: string;
	recruiter?: Recruiter;
	levelId?: string;
	level?: Level;
	categoryPositionId?: string;
	categoryPosition?: CategoryPosition;
	isDeleted?: boolean;
	requirements?: Requirements[];
}

export class PositionFilterModel {
	search?: string;
	userId?: boolean;
	fromSalary?: number;
	toSalary?: number;
	negotiatedSalary?: boolean;
	fromMaxHiringQty?: number;
	toMaxHiringQty?: number;
	fromDate?: Date;
	toDate?: Date;
	stringOfLevelIds?: string;
	stringOfCategoryPositionIds?: string;
	stringOfCompanyIds?: string;
	stringOfLanguageIds?: string;
}

File: BackEnd/Service/Models/PositionModel.cs
namespace Service.Models;

public class PositionModel
{
    public Guid PositionId { get; set; }

    public string? PositionName { get; set; }

    public string? Description { get; set; }

    public string? ImageURL { get; set; }

    public decimal? MinSalary { get; set; }
    public decimal? MaxSalary { get; set; }

    public int MaxHiringQty { get; set; }

    public DateTime? StartDate { get; set; }

    public DateTime? EndDate { get; set; }
    public Guid LevelId { get; set; }
    public LevelModel Level { get; set; }

    public Guid CompanyId { get; set; }
    public CompanyModel? Company { get; set; }

    public Guid LanguageId { get; set; }
    public LanguageModel? Language { get; set; }

    public Guid RecruiterId { get; set; }
    public RecruiterModel Recruiter { get; set; } 

    public Guid CategoryPositionId { get; set; }
    public CategoryPositionModel CategoryPosition { get; set; } 

    public bool IsDeleted { get; set; } = false;

    public virtual ICollection<RequirementModel> Requirements { get; set; } = new List<RequirementModel>();
}